src/RovesODrive.h

```diff
@@ -7,262 +7,302 @@
-
-enum PacketStatus {ValidPacket, InvalidPacket, NoPacket, OverflowPacket};
-enum SerialStatus {SerialGood, SerialFault};
-
-int charToInt(char input[]);
-float charToFloat(char input[]);
-bool charToBool(char input[]);
-void intToChar(char* output, int value);
-void boolToChar(char* output, int value);
-void floatToChar(char* output, int value, uint8_t precision);
-
-void writeODrive(HardwareSerial* mySerial, bool write_request, char* id, char* value, uint8_t axis);
-
-
-
-struct ODrivePacket
+
+//Configs
+#define BRAKE_RESISTANCE				"config.brake_resistance"				//float	
+#define CALIBRATION_CURRENT				"motor.config.calibration_current"		//float
+#define CONTROL_MODE					"controller.config.control_mode"		//int
+#define CURRENT_LIMIT					"motor.config.current_lim"				//float
+#define POLE_PAIRS						"motor.config.pole_pairs"				//int
+#define READ_CURRENT_STATE				"current_state"							//int
+#define VELOCITY_LIMIT					"controller.config.vel_limit"			//float
+#define WRITE_CURRENT_STATE				"requested_state"						//int
+#define WATCHDOG						"config.watchdog_timeout"				//int
+#define WATCHDOG_UPDATE					"u"										//void
+
+//Controller
+#define CURRENT_SETPOINT				"controller.current_setpoint"			//float
+#define VELOCITY_RAMP_ENABLE			"controller.vel_ramp_enable"			//bool
+#define VELOCITY_RAMP_RATE				"controller.config.vel_ramp_rate"		//float
+#define VELOCITY_RAMP_TARGET			"controller.vel_ramp_target"			//float
+#define VELOCITY_SETPOINT				"controller.vel_setpoint"				//float
+
+//Encoder 
+#define CPR								"encoder.config.cpr"					//float
+#define CPR_SETPOINT					"controller.config.setpoints_in_cpr"	//float
+#define POS_CPR							"encoder.pos_cpr"						//float
+#define POS_ESTIMATE					"encoder.pos_estimate"					//float
+
+//Errors
+#define AXIS_ERRORS						"error"									//Error_Axis
+#define CONTROLLER_ERRORS				"controller.error"						//Error_Controller
+#define ENCODER_ERRORS					"encoder.error"							//Error_Encoder
+#define MOTOR_ERRORS					"motor.error"							//Error_Motor
+
+//Gains
+#define POSITION_GAIN					"controller.config.pos_gain"			//float
+#define VELOCITY_GAIN					"controller.config.vel_gain"			//float
+#define VELOCITY_iNTEGRATOR_GAIN		"controller.config.vel_integrator_gain" //float
+
+//Motor Control
+#define MOTOR_CURRENT					"c"										//float
+#define MOTOR_POS						"p"										//float, float, float
+#define MOTOR_TRAJ						"t"										//float
+#define MOTOR_VEL						"v"										//float
+
+//System Commands
+#define ERASE_CONFIG					"se"									//void
+#define SAVE_CONFIG						"ss"									//void
+#define SYSTEM_REBOOT					"sr"									//void
+
+//Trap Trajectory
+#define TRAP_ACCELERATION_LIMIT			"trap_traj.config.accel_limit" 			//float
+#define TRAP_ACCELERATION_PER_COUNTS	"trap_traj.config.A_per_css" 			//float
+#define TRAP_DECELERATION_LIMIT			"trap_traj.config.decel_limit" 			//float
+#define TRAP_VELOCITY_LIMIT  			"trap_traj.config.vel_limit" 			//float
+
+enum Axis_State
```

Comment:
Go through all of the ODrive hpp files, and make sure that your error, and state eumerations are correct.